blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,670 @@
+Specifically, assuming each interface has its own IP address, then each would
+need to be treated as a separate cache server. This means that getting clients
+to use one when the other is unavailable requires manual intervention.
+
+## Proposed Change
+Instead, cache servers should be configurable with numerous network interfaces
+monitored by Traffic Monitor that are all associated with the server, each
+having its own associated network information e.g. IP Address, but Traffic
+Router should only be aware of the "main" interface on which the cache server
+listens for connections.
+
+The specific use-case outlined above, obviously, entails some management be done
+on the cache servers to actually support multiple interfaces answering to the
+same "main" IP address and this should be considered outside of
+<abbr title="Apache Traffic Control">ATC</abbr>'s scope of operation.
+
+## Data Model Impact
+The proposed structure of each of these interfaces is given below as a pair of
+TypeScript<sup>[1](#typescript)</sup> interfaces:
+```typescript
+interface IPAddress {
+	/**
+	 * The actual IP (v4 or v6) address which is used by an interface.
+	 * Any "mask" should be present here as a CIDR-notation suffix.
+	 */
+	address: string;
+
+	/**
+	 * The IP (v4 or v6) address of the gateway used by this IP address.
+	 */
+	gateway: string;
+
+	/**
+	 * Tells whether or not this address of this interface is the server's
+	 * "service" address.
+	 * At least one address of EXACTLY ONE interface MUST have this set to
+	 * 'true' for a server.
+	 */
+	serviceAddress: boolean;
+}
+
+interface Interface {
+	/**
+	 * The name of the interface device on the server e.g. eth0.
+	 */
+	name: string;
+	/**
+	 * These will be all of the IPv4/IPv6 addresses assigned to the interface,
+	 * including gateways and "masks".
+	 * It is illegal for an interface to not have at least one associated IP
+	 * address.
+	 */
+	ipAddresses: Array<IPAddress> & {0: IPAddress};
+	/**
+	 * Whether or not Traffic Monitor should monitor this particular interface.
+	 */
+	monitor: boolean;
+	/**
+	 * The interface's Maximum Transmission Unit.
+	 * If this is 'null' it is assumed that the interface's MTU is not known/is
+	 * irrelevant.
+	 */
+	mtu: 1500 | 9000 | null;
+}
+```
+We don't have a real data model for ATC, so what follows is an approximately
+defined representation of a "server" that is valid within certain contexts. The
+modifications made to it for the purposes of the changes herein proposed should
+be considered valid in those same contexts - whatever they may be.
+
+```typescript
+interface Server {
+
+	cachegroup?:  string | null;
+	cachegroupId: number;
+	cdnId:        number;
+	cdnName?:     string | null;
+	domainName:   string;
+	guid?:        string | null;
+	hostName:     string;
+	httpsPort?:   number | null;
+	id?:          number;
+
+	// ILO things aren't being moved in because they also require a
```

Comment:
We might consider changing the nomenclature around iLo to OOB as that's an HP specific term for an out-of-band management interface.

Reply:
I agree, but that's out of scope for this blueprint.